File: problems/645-set-mismatch/set-mismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const findErrorNums = (nums) => {
    let result = [];
    const count = {};
    for (const n of nums) {
        count[n] = (count[n] || 0) + 1;
    }

    for (let i = 1; i <= nums.length; i++) {
        if (count[i] > 1) {
            result[0] = i;
        }
        if (!count[i]) {
            result[1] = i;
        }
    }
    return result;
};

File: problems/645-set-mismatch/set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        result = [0,0]
        countNums = {}

        for n in nums:
            if n in countNums:
                countNums[n] = countNums[n] + 1
                result[0] = n
            else:
                countNums[n] = 1
        
        for i in range(len(nums)):
            num = i + 1
            if num not in countNums:
                result[1] = num
        
        return result
